### ISSUE TITLE
[a11y]: Date picker examples and Figma assets rely on a placeholder or inconsistent date masks

### ISSUE THREAD

**Issue by mbgower** [carbon-design-system/carbon#11805]
### Package

@carbon/react

### Browser

Chrome

### Operating System

MacOS

### Package version

https://carbondesignsystem.com/components/date-picker/usage

### React version

https://carbondesignsystem.com/components/date-picker/usage

### Automated testing tool and ruleset

n/a

### Description

Carbon has some clearly stated requirement in the Usage material for Date picker, as follows:

<img width="805" alt="Screenshot 2022-07-14 at 9 59 45 AM" src="https://user-images.githubusercontent.com/14298245/179039737-2ab2c70b-ee0d-46e1-8399-85c51365db5d.png">

Despite this, Carbon's illustrations for date picker seem to default to using only the placeholder text for date mask info. Date masks as placeholders -- or other variants -- appear through the Usage and Style info in Carbon's documentation:

<img width="588" alt="image" src="https://user-images.githubusercontent.com/14298245/179041995-774ea815-f6c7-43c4-8c86-d6ec2509686b.png">
^ In the Live demo

<img width="924" alt="image" src="https://user-images.githubusercontent.com/14298245/179042157-94955b29-44de-499c-abc5-5c9e7c6e0eb5.png">
^ In examples 

<img width="663" alt="image" src="https://user-images.githubusercontent.com/14298245/179043302-f0b26278-d6c4-438f-97b6-fc9ca30b32f4.png">
^ On the style page

<img width="816" alt="image" src="https://user-images.githubusercontent.com/14298245/179043435-13fd83e5-5132-43f9-9c13-dd9650c319b0.png">
^ In the Date input specs

<img width="751" alt="image" src="https://user-images.githubusercontent.com/14298245/179043578-ea229efa-3a71-4cf8-923f-17e46851389b.png">
^ In the size specs

<img width="490" alt="image" src="https://user-images.githubusercontent.com/14298245/179043773-1eb50f8b-24e3-46b0-a36a-aa978e9a5bf8.png">
^in the React storybook

 In Figma, every example shows it without it part of the label or helper text
![image](https://user-images.githubusercontent.com/14298245/179041362-72342970-2633-459e-b4c7-82acfc4753aa.png)

--- 

Where Carbon _does_ show a mask, it tends to be inconsistent:
<img width="284" alt="image" src="https://user-images.githubusercontent.com/14298245/179042876-e10b6238-ed2f-45a6-9cc4-c714b653eb4c.png">
In a _different_ position, in the States material

<img width="244" alt="image" src="https://user-images.githubusercontent.com/14298245/179043140-1b014501-6164-444f-a595-b4b915032202.png">
^ In a different position (and format) on the Light variant example
---

Note that in the quoted text at the top, Carbon also says:
> When using calendar picker the date format will be automated if the user selects from the calendar menu.

This implies it is okay to have the mask in the placeholder here, but it's not really valid. First, at the moment the ONLY way for a keyboard user to easily changed date and month is to manually enter a date; so the date mask remains a problem) even where the calendar is offered.
Second, there are two very widespread and contradictory ways of showing month/day combinations (MM/DD/YYYY and DD/MM/YYYY).

Is 7/8/2022 the seventh of August or the eighth of July? Without the mask, users cannot tell, so half your audience is going to be wrong in their assumption.

Outside the confusion for anyone trying to interpret a date once the placeholder date mask info has been replaced by an actual date, the placeholder causes several challenges specific to accessibility:
1. The placeholder text has very low contrast, and may be difficult to read by anyone with low vision
2. The placeholder text is not surfaced consistently by screen readers and other ATs.

Carbon needs to have a consistent default presentation of date mask which is adopted across its prototyping tools, coding repos and documentation.

___

Note, this and https://github.com/carbon-design-system/carbon/issues/11800 have been separated from [less critical date picker considerations/issues](https://github.com/carbon-design-system/carbon-website/issues/3045) because they directly affect the accessibility guidance that can be crafted. As such, they constitute higher priority considerations.

### Reproduction/example

https://carbondesignsystem.com/components/date-picker/usage

### Steps to reproduce

Go to the Figma library, react repo or Carbon documentation site and look.

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/carbon-design-system/carbon/blob/f555616971a03fd454c0f4daea184adf41fff05b/.github/CODE_OF_CONDUCT.md)
- [X] I checked the [current issues](https://github.com/carbon-design-system/carbon/issues) for duplicate problems

**Comment by jnm2377** [comment]
I think what we can do to address this is:
1. update our storybook example to have `helperText` for simple datepickers
2. make the `helperText` prop default value be the date format for `type="simple"` date pickers

**Comment by mbgower** [comment]
@jnm2377 those two technical changes will definitely help. 
As discussed, there are also design considerations here, which I'm hoping @laurenmrice and @aagonzales can lead, to do with both the visual examples in Carbon documentation, and the prototyping assets in Figma and Sketch.

**Comment by mbgower** [comment]
My recommendation would be to have the default version of all date pickers have the date mask present in the Helper text by default. This will arguably improve the experience for many users.

As discussed in the Date picker usage.pdf, I'm not certain how/if the OS-level date preferences affect anything in Carbon (i.e., if I have my system set to dd/mm/yyyy, does the browser inherit this and alter Carbon's default of mm/dd/yyyy?). The technical details should be considered as part of this decision.

**Comment by mbgower** [comment]
@sstrubberg  I see you folks have put it in the Design System backlog. I agree this is pretty much a design issue. Hopefully @laurenmrice and team will review this.
